vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,226 @@
+	// Provided VM IDs must be part of the solution being transitioned and
+	// must exist in the cluster where the solution is installed.
+	// Provided VmSelectionSpecs must be present in the applied desired state as part of the
+	// ClusterSolutionSpec#alternativeVmSpecs.
+	//
+	// If unset, no VmSelectionSpecs are applied on the source agencies' System VMs during the enablement.
+	SourceVmSelectionSpecs map[string]*VmSelectionSpec `json:"source_vm_selection_specs,omitempty"`
+
+	// ClusterModule to be reused for transitioned System VMs. Used to express
+	// VM-VM anti affinity relation between System VMs in the vSphere Cluster.
+	// The module must exist for the cluster where the solution is installed.
+	//
+	// If unset, no cluster module is reused. vLCM creates a new module if needed.
+	ClusterModule string `json:"cluster_module,omitempty"`
+}
+
+// Enable enables an EAM managed solution in vLCM. The solution specification is
+// validated before the enablement is started.
+//
+// The enablement only transfers ownership of the solution from EAM to LCCM
+// and sets the desired state in LCCM. The new desired state is not applied,
+// the solution system VMs are untouched.
+//
+// The following happens once the operation is started:
+//   - A removal of the corresponding agency in EAM is triggered.
+//
+// The following happens once the operation is completed:
+//   - The corresponding agency in EAM can no longer be controlled through the EAM API.
+//   - The management of the desired solution specification can be done only through vLCM. See Solutions
+//
+// Parameters:
+//   - cluster: Identifier of the cluster.
+//   - solution: Identifier of the solution.
+//   - spec: Enablement specification.
+//
+// Returns an error if:
+//   - There is an internal error. The accompanying error message will give more details about the failure.
+//   - InvalidArgument: If the validation of the cluster desired specification fails, the solution associated with solution already exists in vLCM for the cluster, or the agency associated with the eamAgencyID exists, but is not in the scope of the cluster. The value of the data field of Error contains more information. It is a structure that contains all the fields defined in ValidationResult.
+//   - NotAllowedInCurrentState: If System VMs are disabled on the cluster via internal ESX Agent Manager (EAM) API (eam.EsxtAgentManager#disable).
+//   - Unsupported: If the cluster associated with cluster is not managed by vLCM.
+//   - NotFound: If there is no cluster associated with the cluster or if there is no EAM agency associated with the eamAgencyID field of the spec in the system.
+//   - ServiceUnavailable: If the service is not available.
+//   - Unauthenticated: If the caller is not authenticated.
+//   - Unauthorized: If the user doesn't have the required privileges.
+func (m *Manager) Enable(ctx context.Context, cluster types.ManagedObjectReference, solution string, spec *EnableSpec) error {
```

Comment:
Make this a private function, ex. `func (m *Manager) enable` with two public version:

* `func (m *Manager) Enable`
* `func (m *Manager) EnableAsync`

They both call the private one, but the second one returns the task instead of blocking.